policy/modules/system/systemd.if

```diff
@@ -58,28 +61,691 @@ template(`systemd_role_template',`
+	allow $1_systemd_t systemd_user_runtime_t:file manage_file_perms;
+	allow $1_systemd_t systemd_user_runtime_t:fifo_file manage_fifo_file_perms;
+	allow $1_systemd_t systemd_user_runtime_t:lnk_file manage_lnk_file_perms;
+	allow $1_systemd_t systemd_user_runtime_t:sock_file manage_sock_file_perms;
+
 	selinux_use_status_page($1_systemd_t)
 
 	seutil_read_file_contexts($1_systemd_t)
 	seutil_search_default_contexts($1_systemd_t)
+
+	systemd_list_journal_dirs($3)
+	systemd_read_journal_files($3)
+
+	init_list_unit_dirs($1_systemd_t)
+
+	systemd_manage_conf_home_content($3)
+	systemd_relabel_conf_home_content($3)
+
+	systemd_manage_data_home_content($3)
+	systemd_relabel_data_home_content($3)
+
+	systemd_search_user_runtime_unit_dirs($1_systemd_t)
+	allow $1_systemd_t systemd_user_runtime_unit_t:dir manage_dir_perms;
+	allow $1_systemd_t systemd_user_runtime_unit_t:file manage_file_perms;
+	allow $1_systemd_t systemd_user_runtime_unit_t:lnk_file manage_lnk_file_perms;
+
+	systemd_read_user_unit_files($3)
+	systemd_read_user_unit_files($1_systemd_t)
+
+	systemd_search_user_runtime_unit_dirs($1_systemd_t)
+	systemd_list_user_runtime_unit_dirs($3)
+	systemd_read_user_runtime_units($3)
+
+	systemd_status_user_units($3)
+	systemd_start_user_units($3)
+	systemd_stop_user_units($3)
+	systemd_reload_user_units($3)
+	systemd_status_conf_home_units($3)
+	systemd_start_conf_home_units($3)
+	systemd_stop_conf_home_units($3)
+	systemd_reload_conf_home_units($3)
+
+	type_transition systemd_user_session_type systemd_user_runtime_t:dir systemd_user_runtime_unit_t "generator.early";
+	type_transition systemd_user_session_type systemd_user_runtime_t:dir systemd_user_runtime_unit_t "generator.late";
+	type_transition systemd_user_session_type systemd_user_runtime_t:dir systemd_user_runtime_unit_t "transient";
+	type_transition systemd_user_session_type systemd_user_runtime_t:dir systemd_user_runtime_unit_t "user";
+
+	init_dbus_chat($3)
+	systemd_status_user_runtime_units($3)
+	systemd_start_user_runtime_units($3)
+	systemd_stop_user_runtime_units($3)
+	systemd_reload_user_runtime_units($3)
+
+	optional_policy(`
+		xdg_config_filetrans($1_systemd_t, systemd_conf_home_t, dir, "systemd")
+		xdg_data_filetrans($1_systemd_t, systemd_data_home_t, dir, "systemd")
+		xdg_read_config_files($1_systemd_t)
+		xdg_read_data_files($1_systemd_t)
+	')
+')
+
+######################################
+## <summary>
+##   Allow the specified domain to communicate with the specified
+##   dbusd domain. Should only be used for user systemd types.
+## </summary>
+## <param name="domain">
+##   <summary>
+##     Prefix for the user domain.
+##   </summary>
+## </param>
+## <param name="dbusd_domain">
+##   <summary>
+##     dbusd domain to allow interactions with.
+##   </summary>
+## </param>
+#
+interface(`systemd_user_spec_daemon_domain',`
+	gen_require(`
+		type $1_systemd_t;
+	')
+
+	dbus_spec_domtrans($1_systemd_t, $2)
+	dbus_spec_session_bus_client($1, $1_systemd_t)
```

Comment:
this does not belong here it should probably go into the `systemd_role_template`